controllers/scopetemplate_controller_test.go

```diff
@@ -0,0 +1,373 @@
+					clusterRoleBindingGenerateKey: "test"}
+
+				roleBindingList := listRoleBinding(namespace.GetName(), 1, labels)
+
+				existingRB := &roleBindingList.Items[0]
+
+				verifyRoleBindings(existingRB, scopeInstance, scopeTemplate)
+			})
+
+			When("a scopeInstance is updated to include another namespace", func() {
+				BeforeEach(func() {
+					namespace2 = &corev1.Namespace{
+						ObjectMeta: metav1.ObjectMeta{
+							GenerateName: "test-",
+						},
+					}
+					Expect(k8sClient.Create(ctx, namespace2)).NotTo(HaveOccurred())
+				})
+
+				AfterEach(func() {
+					Expect(k8sClient.Delete(ctx, namespace2)).NotTo(HaveOccurred())
+				})
+
+				It("Should create new roleBindings in the expected namespaces", func() {
+					Eventually(func() error {
+						scopeInstance := &operatorsv1.ScopeInstance{}
+						if err := k8sClient.Get(ctx, types.NamespacedName{Name: scopeInstanceName}, scopeInstance); err != nil {
+							return err
+						}
+						scopeInstance.Spec.Namespaces = []string{namespace.GetName(), namespace2.GetName()}
+						if err := k8sClient.Update(ctx, scopeInstance); err != nil {
+							return err
+						}
+						return nil
+					}, timeout, interval).Should(BeNil())
+
+					labels := map[string]string{scopeInstanceUIDKey: string(scopeInstance.GetUID()),
+						scopeTemplateUIDKey:           string(scopeTemplate.GetUID()),
+						clusterRoleBindingGenerateKey: "test"}
+
+					roleBindingList := listRoleBinding(namespace2.GetName(), 1, labels)
+
+					existingRB := &roleBindingList.Items[0]
+
+					verifyRoleBindings(existingRB, scopeInstance, scopeTemplate)
+
+					roleBindingList = listRoleBinding(namespace.GetName(), 1, labels)
+
+					existingRB = &roleBindingList.Items[0]
+					verifyRoleBindings(existingRB, scopeInstance, scopeTemplate)
+				})
+
+				When("a scopeInstance is updated to remove one of the namespace", func() {
+					It("Should remove respective roleBindings in the expected namespaces", func() {
+						Eventually(func() error {
+							scopeInstance := &operatorsv1.ScopeInstance{}
+							if err := k8sClient.Get(ctx, types.NamespacedName{Name: scopeInstanceName}, scopeInstance); err != nil {
+								return err
+							}
+							scopeInstance.Spec.Namespaces = []string{namespace2.GetName()}
+							if err := k8sClient.Update(ctx, scopeInstance); err != nil {
+								return err
+							}
+							return nil
+						}, timeout, interval).Should(BeNil())
+
+						labels := map[string]string{scopeInstanceUIDKey: string(scopeInstance.GetUID()),
+							scopeTemplateUIDKey:           string(scopeTemplate.GetUID()),
+							clusterRoleBindingGenerateKey: "test"}
+
+						roleBindingList := listRoleBinding(namespace2.GetName(), 1, labels)
+
+						existingRB := &roleBindingList.Items[0]
+
+						verifyRoleBindings(existingRB, scopeInstance, scopeTemplate)
+
+						roleBindingList = listRoleBinding(namespace.GetName(), 0, labels)
+
+						Expect(len(roleBindingList.Items)).To(Equal(0))
```

Comment:
You do not need to check the length of the list if you are already checking it in the `listRoleBinding` func.

Reply:
right, removed it.